### PR TITLE
add Cognite Data Fusion

### PR DESCRIPTION
**Pull request by sceniclife**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

### CODE REVIEW

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 1]
```diff
@@ -5,31 +5,177 @@
     ],
     "connectionParameters": {
       "token": {
-        "type": "oauthSetting",
+        "type": "oAuthSetting",
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json]: Can you please refer the schema link below and change the word as "oauthSetting"

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 2]
```diff
@@ -5,31 +5,177 @@
...
         "oAuthSettings": {
-          "identityProvider": "oauth2",
+          "identityProvider": "oauth2generic",
           "clientId": "PLACEHOLDER_CLIENTID",
-          "scopes": ["https://bluefield.cognitedata.com/.default"],
+          "scopes": ["{Scope}"],
           "redirectMode": "Global",
-          "redirectUrl": "https://global.consent.azure-apim.net/redirect",
+          "redirectUrl": "{RedirectUrl}",
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json]: redirectUrl is not needed to mention here.  Team will guide you to add your client app during certification process. Please remove this

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 3]
```diff
@@ -5,31 +5,177 @@
...
+        "uiDefinition": {
+          "constraints": {
+            "required": "true"
+          },
+          "displayName": "Enter your Cognite Data Fusion Cluster.",
+          "description": "Enter your Cognite Data Fusion Cluster.",
+          "tooltip": "api, greenfield, bluefield, az-eastus-1, etc..."
+        }
+      },
+      "token:TenantId": {
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json:59]: Please fix the below error, and refer the below link for reference (line no. 160). 

##[error]@{level=Error; message=Required properties are missing from object: metadata.; valuePath=line: 60 position: 25 path: '[0].properties.connectionParameters.token:TenantId'; value=System.Object[]}


https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 4]
```diff
@@ -5,31 +5,177 @@
...
+        "type": "string",
+        "uiDefinition": {
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json:64]: Please fix below errors. 

##[error]@{level=Critical; message=Missing metadata.; valuePath=$.properties.connectionParameters.token:TenantId.metadata; value=}

##[error]@{level=Critical; message=Missing hidden.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.constraints; value=}

##[error]@{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=Enter your Tenant ID.}

##[error]@{level=Critical; message=Invalid value for the display name property within the UI Defintion object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.displayName; value=Enter your Tenant ID.}

Please refer the below link for reference. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json#L214

**Review by vmanoharas — COMMENTED** [review]
Hello @sceniclife, 

Please fix the errors added as comments.

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 5]
```diff
@@ -7,29 +7,167 @@
       "token": {
         "type": "oauthSetting",
         "oAuthSettings": {
-          "identityProvider": "oauth2",
+          "identityProvider": "oauth2generic",
           "clientId": "PLACEHOLDER_CLIENTID",
-          "scopes": ["https://bluefield.cognitedata.com/.default"],
+          "scopes": ["{Scope}"],
           "redirectMode": "Global",
           "redirectUrl": "https://global.consent.azure-apim.net/redirect",
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json]: As I mentioned earlier, could you kindly remove this redirectUrl completely, as this is not needed to add here and change redirectMode value from "Global" to "GlobalPerConnector".  Please refer this link. https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-5-submit-your-connector-for-deployment

**Review by vmanoharas — COMMENTED** [review]
Hello @sceniclife, 

Could you please resolve the review comments?

File: certified-connectors/Cognite Data Fusion/apiProperties.json (hunk 5 above)
Comment (sceniclife) [reply certified-connectors/Cognite Data Fusion/apiProperties.json]: When I change redirectMode, I get this:

> Value is not accepted. Valid values: "Global", "Direct".

Comment (sceniclife) [reply certified-connectors/Cognite Data Fusion/apiProperties.json]: Do I leave `redirect_uri={redirectUrl}` in my templates? or just remove them?

File: certified-connectors/Cognite Data Fusion/apiProperties.json [hunk 6]
```diff
@@ -7,29 +7,166 @@
...
+        "metadata": {
+          "sourceType": "AzureActiveDirectoryTenant"
+        },
+        "type": "string",
+        "uiDefinition": {
+          "constraints": {
+            "hidden": "true",
+            "required": "false"
+          },
+          "description": "The tenant ID of for the Azure Active Directory application",
```
Comment (vmanoharas) [inline certified-connectors/Cognite Data Fusion/apiProperties.json]: The apiproperties.json validation is still the error. Could you please remove description and display properties?

##[error]@{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=The tenant ID of for the Azure Active Directory application}
##[error]@{level=Critical; message=Invalid value for the display name property within the UI Defintion object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.displayName; value=Tenant}

**Review by vmanoharas — COMMENTED** [review]
Hello @sceniclife, 

The validation check still showing the error. can you please look into my comment and resolve it?

**Review by vmanoharas — APPROVED** [review]
Hello @sceniclife 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Comment by vmanoharas** [comment]
Hello @sceniclife, 

Can you please upload artifacts here in zip format?

**Pull request merged**